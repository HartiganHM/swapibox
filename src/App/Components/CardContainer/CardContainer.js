import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css';

const CardContainer = () => {
    return(
        <div className='card-container'>
            <span className='placeholder'>Select a Category</span>
            <Card />
        </div>
    )
}

export default CardContainer;