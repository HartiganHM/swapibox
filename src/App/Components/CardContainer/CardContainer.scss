@import '../../../variables.scss';

.card-container {
    background-color: $background-dark;
    color: $background-light;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    text-align: left;
    padding: $padding-small;

    .placeholder {
        align-self: center;
        font-size: 3.2rem;
        height: -webkit-fill-available;
        margin-top: 3rem;
    }
}