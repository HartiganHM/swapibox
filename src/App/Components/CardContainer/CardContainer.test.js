import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer Tests', () => {
  let renderedCardContainer;
  let mockCategory;
  let mockCurrentDisplay;
  let mockCurrentFavorites;
  let mockSaveFavorites;
  let mockRemoveFavorites;

  beforeEach(() => {
    mockCategory = [];
    mockCurrentDisplay = null;
    mockCurrentFavorites = [];
    mockSaveFavorites = jest.fn();
    mockRemoveFavorites = jest.fn();

    renderedCardContainer = shallow(
      <CardContainer
        category={mockCategory}
        currentDisplay={mockCurrentDisplay}
        currentFavorites={mockCurrentFavorites}
        saveFavorite={mockSaveFavorites}
        removeFavorite={mockRemoveFavorites}
      />
    );
  });

  it('Should exist', () => {
    expect(renderedCardContainer).toBeDefined();
  });

  it('Should match the snapshot', () => {
    expect(renderedCardContainer).toMatchSnapshot();
  });

  it('Should render Cards for each object in the category', () => {
    expect(renderedCardContainer.find('Card').length).toEqual(0);

    mockCategory = [{ name: 1 }, { name: 2 }, { name: 3 }];
    renderedCardContainer = shallow(<CardContainer category={mockCategory} />);

    expect(renderedCardContainer.find('Card').length).toEqual(3);
  });

  it('Should change message if the currentDisplay is favorites', () => {
      expect(renderedCardContainer.find('.placeholder').text()).toEqual('Select a category');

      mockCurrentDisplay = 'favorites';
      renderedCardContainer = shallow(<CardContainer currentDisplay={mockCurrentDisplay} />);

      expect(renderedCardContainer.find('.placeholder').text()).toEqual('No favorites to display');
  });

  it('Should not display a placeholder message if there are cards to display', () => {
      expect(renderedCardContainer.find('.placeholder').length).toEqual(1);

      mockCategory = [{ name: 1 }, { name: 2 }, { name: 3 }];
      renderedCardContainer = shallow(<CardContainer category={mockCategory} />);

      expect(renderedCardContainer.find('.placeholder').length).toEqual(0);
  });
});
