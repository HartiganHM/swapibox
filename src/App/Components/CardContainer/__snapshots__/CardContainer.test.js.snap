// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    category={Array []}
    currentDisplay={null}
    currentFavorites={Array []}
    removeFavorite={[Function]}
    saveFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="placeholder"
>
        Select a category
</span>,
      "className": "card-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Select a category",
        "className": "placeholder",
      },
      "ref": null,
      "rendered": "Select a category",
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="placeholder"
>
          Select a category
</span>,
        "className": "card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a category",
          "className": "placeholder",
        },
        "ref": null,
        "rendered": "Select a category",
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
